### DISCUSSION TITLE
Decrease of density in IISPH

### DISCUSSION THREAD

**Discussion by nimeeshrathi** [InteractiveComputerGraphics/SPlisHSPlasH#294] [Q&A]
Hi Jan,

Hope you're doing well. I'm currently working on SPHERIC test case 2( the dam break with an obstacle). I could use your guidance on a couple of points:

1. **Density Decrease Issue**:
   I'm using the IISPH solver, and I've noticed that the density is decreasing by more than 10% of the rest density. Any advice on how to prevent this for a more accurate result would be greatly appreciated. ( I uses Max density error : 0.05 )

2. **Water Propagation Issue:**
   When the water first hits the obstacle, a significant amount of it is not moving forward as expected. Instead, it seems to propagate backward. I've experimented with different viscosity values and CFL factors, but I'm still puzzled as to why this is happening.

Here are some additional details:
   - Viscosity: XSP with a value of 0.3
   - Boundary Handling: Volume maps
   - Dam map resolution: 236*76*68
   - Obstacle map resolution: 403*161*161
   - CFL factor : 0.1
.

Please find the attached videos for more info 
[Dam_break.zip](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/13412650/Dam_break.zip)

Pink particles are ones with a density below 900, and white particles have a density above 900

Any insights or suggestions you have would be really helpful.

**Comment by janbender** [comment]
1. The max density error of 0.05 says that the density error should be at max 5 percent. However, the solver only cares for the error rho^* - rho0. This means the advected denstiy minus the rest density. To improve this you can reduce the time step size. 

2. What do you mean with "moving forward"? If there is an obstacle, the fluid cannot move forwards.
